packages/build-ui/src/pages/controller.rs

```diff
@@ -0,0 +1,142 @@
+        let user: UserService = use_context();
+        let ctrl = Controller { lang, popup, user };
+        Ok(ctrl)
+    }
+
+    pub fn open_select_blockchain_modal(&self) {
+        let mut popup = self.popup.clone();
+        let tr: BlockchainSelectionModalTranslate = translate(&self.lang);
+        let ctrl = self.clone();
+        popup
+            .open(rsx! {
+                BlockchainSelectionModal {
+                    lang: self.lang,
+                    on_select: move |chain| {
+                        tracing::debug!("Selected blockchain: {:?}", chain);
+                        ctrl.open_connect_wallet_modal(chain);
+                    },
+                }
+            })
+            .with_id("select_blockchain_modal")
+            .with_title(tr.title);
+    }
+
+    pub fn open_connect_wallet_modal(&self, chain: Chain) {
+        let mut popup = self.popup.clone();
+        let tr: ConnectWalletModalTranslate = translate(&self.lang);
+        let mut ctrl = self.clone();
+
+        popup
+            .open(rsx! {
+                ConnectWalletModal {
+                    lang: self.lang,
+                    on_select: move |wallet: Wallet| {
+                        tracing::debug!("Selected wallet: {:?}", wallet);
+                        spawn(async move {
+                            if let Ok(status) = ctrl.user.login(chain, wallet).await {
+                                tracing::debug!("Login status: {:?}", status);
+                                match status {
+                                    Status::Login => {
+                                        popup.close();
+                                    }
+                                    Status::Signup { principal: _, profile_url: _, nickname, email } => {
+                                        ctrl.open_user_info_modal(nickname, email);
+                                    }
+                                    _ => {
+                                        popup.close();
+                                    }
+                                }
+                            }
+                        });
+                    },
+                }
+            })
+            .with_id("connect_wallet_modal")
+            .with_title(tr.title);
+    }
+
+    // // New method that takes the selected blockchain and wallet
+    pub fn open_user_info_modal(&self, nickname: Option<String>, email: Option<String>) {
+        let mut popup = self.popup.clone();
+        let tr: UserInfoModalTranslate = translate(&self.lang);
+        let mut ctrl = self.clone();
+        popup
+            .open(rsx! {
+                UserInfoModal {
+                    nickname,
+                    email,
+                    lang: self.lang,
+                    on_button_click: move |result: UserInfoResult| {
+                        tracing::debug!("Selected wallet: {:?}", result);
+                        spawn(async move {
```

Comment:
Why did you need to spawn?

```suggestion
                    on_button_click: move |result: UserInfoResult| async move {
                        tracing::debug!("Selected wallet: {:?}", result);```

Reply:
Simply made a mistake. Spawning is not necessary